internal/clientcache/internal/cache/refresh.go

```diff
@@ -131,6 +144,85 @@ func (r *RefreshService) refreshableUsers(ctx context.Context, in []*user) ([]*u
 	return ret, nil
 }
 
+// RefreshForSearch refreshes a specific resource type owned by the user associated
+// with the provided auth token id, if it hasn't been refreshed in a certain
+// amount of time. If the resources has been updated recently, or if it is known
+// that the user's boundary instance doesn't support partial refreshing, this
+// method returns without any requests to the boundary controller.
+// While the criteria of whether the user will refresh or not is almost the same
+// as the criteria used in Refresh, RefreshForSearch will not refresh any
+// data if there is not at least 1 refresh token stored for the resource. It
+// might make sense to change this in the future, but the reasoning used is
+// that we should not be making an initial load of all resources while blocking
+// a search query, in case we have not yet even attempted to load the resources
+// for this user yet.
+func (r *RefreshService) RefreshForSearch(ctx context.Context, authTokenid string, resourceType SearchableResource, opt ...Option) error {
+	const op = "cache.(RefreshService).RefreshForSearch"
+	if r.maxSearchRefreshTimeout > 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, r.maxSearchRefreshTimeout)
```

Comment:
I thought a bit more about this, and it would be really cool if instead of canceling the refresh operation completely this would simply return early and read from the DB while letting the refresh operation continue to run in the background. It would be a little more complicated as you'd essentially need to start a goroutine that writes to the DB and the send a signal on a channel, where this goroutine could wait on that channel or this context to timeout. Anyway, that would mean that the update could still take place, but asynchronously, if the user has waited long enough already. What do you think? It could also be something we do later, not necessarily something we do now.